### PR TITLE
fix: Check permissions for team OOO entries

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Fixes CAL-3527

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Please refer to CAL-3527 ticket in Linear

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts [hunk 1]
```diff
@@ -50,6 +50,14 @@ export const outOfOfficeCreate = async ({ ctx, input }: TBookingRedirect) => {
...
       where: {
         id: input.toTeamUserId,
+        /** You can only create OOO for members of teams you're either ADMIN or OWNER */
+        teams: {
+          some: {
+            userId: ctx.user.id,
+            accepted: true,
+            role: { in: ["OWNER", "ADMIN"] },
+          },
+        },
```
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:60]: You will receive an obfuscated not found error if you try to create entries for other users.

File: packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts [hunk 2]
```diff
@@ -194,28 +202,18 @@ export const outOfOfficeEntryDelete = async ({ ctx, input }: TBookingRedirectDel
     throw new TRPCError({ code: "BAD_REQUEST", message: "out_of_office_id_required" });
   }
 
-  // Validate outOfOfficeEntry belongs to the user deleting it
-  const outOfOfficeEntry = await prisma.outOfOfficeEntry.findFirst({
-    select: {
-      uuid: true,
-      userId: true,
-    },
+  const deletedOutOfOfficeEntry = await prisma.outOfOfficeEntry.delete({
```
Comment (zomars) [inline packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:205]: Refactored so this runs in a single query

**Review by zomars — COMMENTED** [review]
Self review done

**Review by emrysal — DISMISSED** [review]

File: packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts [hunk 3]
```diff
@@ -50,6 +50,14 @@ export const outOfOfficeCreate = async ({ ctx, input }: TBookingRedirect) => {
     const user = await prisma.user.findUnique({
       where: {
         id: input.toTeamUserId,
+        /** You can only create OOO for members of teams you're either ADMIN or OWNER */
+        teams: {
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:54]: shouldn't we rather check here if `ctx.user.id` and `toTeamUserId` are in the same teams
?

Comment (zomars) [reply packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:54]: RN we're checking if `toTeamUserId` belongs to a team where the `ctx.user.id` is either an ADMIN or OWNER. Isn't that what we want?

Comment (zomars) [reply packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:54]: I see. Then we can either add MEMBER to the role check or skip the role check altogether.

Comment (zomars) [reply packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:54]: I see. Then we can either add MEMBER to the role check or skip the role check altogether.

Comment (zomars) [reply packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts:54]: Addressed here [b9ac480](https://github.com/calcom/cal.com/pull/14736/commits/b9ac480c42c0664a6f6388621ace4a0f01fe171f)

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**